src/ui/public/vis/agg_configs.js

```diff
@@ -114,6 +114,15 @@ export default function AggConfigsFactory(Private) {
       const dsl = dslLvlCursor[config.id] = config.toDsl();
       let subAggs;
 
+      ((function parseParentAggs(dsl) {
+        if (dsl.parentAggs) {
+          _.each(dsl.parentAggs, (agg, key) => {
+            dslLvlCursor[key] = agg;
+            parseParentAggs(agg);
+          });
+        }
+      })(dsl));
+
```

Comment:
This is imho an advanced construct: a self-invoking function, which is named so it can accommodate recursion.  Consider moving `parseParentAggs` outside the scope and just calling `parseParentAggs(dsl)`. Anyway, that's up to you ;).